### PR TITLE
Remove explicit unicode from strings

### PR DESCRIPTION
**Pull request by JacobCallahan**
Since we are now exclusively python 3, we can remove these.

### CODE REVIEW

**Comment by mirekdlugosz** [comment]
Looks like you missed some:
```bash
$ set +o histexpand  # this is needed by bash to not interpret '!'
$ ack -l "(?<![\"'])\bu[\"']"
robottelo/libvirt_discovery.py
robottelo/constants.py
robottelo/vm.py
robottelo/system_facts.py
robottelo/datafactory.py
docs/conf.py
docs/features/ssh.rst
docs/features/commands.rst
setup.py
tests/robottelo/test_assertions.py
tests/robottelo/test_hammer.py
tests/robottelo/test_host_info.py
tests/foreman/ui/test_ldap_authentication.py
tests/foreman/cli/test_remoteexecution.py
tests/foreman/cli/test_computeresource_libvirt.py
tests/foreman/cli/test_repository.py
tests/foreman/cli/test_activationkey.py
```

They seem to generally fall in one of two categories:
1. `u"` (double-quote):
    ```
    robottelo/vm.py
    163:                    u"Failed to fetch domain from provisioning server: {0} ".format(
    ```
2. Things outside of robottelo or tests:
    ```
    docs/conf.py
    16:project = u'Robottelo'
    ```

File: robottelo/helpers.py [hunk 1]
```diff
@@ -688,8 +688,7 @@ def host_provisioning_check(ip_addr):
     :return: ssh command return code and stdout
     """
     result = ssh.command(
-        u'for i in {{1..60}}; do ping -c1 {0} && exit 0; sleep 20;'
-        u' done; exit 1'.format(ip_addr)
+        'for i in {{1..60}}; do ping -c1 {0} && exit 0; sleep 20;' ' done; exit 1'.format(ip_addr)
```
Comment (mirekdlugosz) [inline robottelo/helpers.py]: Any way to not merge split strings back into one line? Or to actually make them single string?
```suggestion
        'for i in {{1..60}}; do ping -c1 {0} && exit 0; sleep 20; done; exit 1'.format(ip_addr)
```

Comment (JacobCallahan) [reply robottelo/helpers.py]: black doesn't combine strings, but i can certainly make the change manually. nice catch!

File: robottelo/vm.py [hunk 2]
```diff
@@ -662,11 +662,10 @@ def execute_foreman_scap_client(self, policy_id=None):
         """
         if policy_id is None:
             result = self.run(
-                u'awk -F "/" \'/download_path/ {print $4}\' '
-                '/etc/foreman_scap_client/config.yaml'
+                'awk -F "/" \'/download_path/ {print $4}\' ' '/etc/foreman_scap_client/config.yaml'
```
Comment (JacobCallahan) [inline robottelo/vm.py]: note

File: tests/foreman/ui/test_smartclassparameter.py [hunk 3]
```diff
@@ -128,27 +128,24 @@ def test_positive_end_to_end(session, puppet_class, sc_params_list):
...
+            'value': '["{0}","{1}","{2}"]'.format(
                 gen_string('alpha'), gen_string('numeric').lstrip('0'), gen_string('html')
             ),
         },
+        {'sc_type': 'hash', 'value': '{0}: {1}'.format(gen_string('alpha'), gen_string('alpha'))},
         {
-            u'sc_type': 'hash',
-            u'value': '{0}: {1}'.format(gen_string('alpha'), gen_string('alpha')),
+            'sc_type': 'yaml',
+            'value': '--- {0}=>{1}'.format(gen_string('alpha'), gen_string('alpha')),
```
Comment (JacobCallahan) [inline tests/foreman/ui/test_smartclassparameter.py]: note. what made these changes?

File: tests/robottelo/test_host_info.py [hunk 4]
```diff
@@ -85,29 +83,27 @@ def test_command_error(self, logger):
...
 
 class GetHostSatVersionTestCase(TestCase):
     """Tests for get_host_sat_version version"""
 
     SSH_RESULT_ERROR = SSHCommandResult(
         stdout=[],
-        stderr=(
-            u'grep: /usr/share/foreman/lib/satellite/version.rb: No such ' u'file or directory'
-        ),
+        stderr=('grep: /usr/share/foreman/lib/satellite/version.rb: No such ' 'file or directory'),
```
Comment (JacobCallahan) [inline tests/robottelo/test_host_info.py]: note

**Comment by JacobCallahan** [comment]
@mirekdlugosz @rochacbruno @mshriver et. al.
Removing Do Not Merge, as I now believe this to be merge ready. I ran the cli content view tests file and got great resulsts. All three failures were unrelated to these changes.
```======= 3 failed, 82 passed, 10 skipped, 6 deselected in 7044.17 seconds =======```

**Comment by mirekdlugosz** [comment]
Hm, my previous test case still doesn't pass:
```
$ ack "(?<![\"'])\bu[\"']"
docs/features/ssh.rst
24:        stdout=[u'Red Hat Enterprise Linux Server release 7.2 (Maipo)', u''],
53:        stdout=[u'Red Hat Enterprise Linux Server release 7.2 (Maipo)', u''],

docs/features/commands.rst
72:    Out[1]: SSHCommandResult(stdout=[u'3.10.0-327.el7.x86_64', u''], stderr='', return_code=0, output_format=None)
105:    2016-09-16 14:01:15 - robottelo.ui.browser - DEBUG - mouseMoveTo:  {'element': u'0.8036987570003233-1'}
108:    2016-09-16 14:01:46 - robottelo.ui.browser - DEBUG - clickElement:  {'id': u'0.12969267888817115-2'}
111:    2016-09-16 14:02:13 - robottelo.ui.browser - DEBUG - sendKeysToElement:  {'id': u'0.12969267888817115-3', 'value': 'my_username'}

tests/foreman/ui/test_ldap_authentication.py
231:        cmd="ipa-client-install --password '{}' --domain {} --server {} --realm {} -U".format(

tests/foreman/cli/test_remoteexecution.py
225:            '''stat -c '%U' /home/{0}/{1}'''.format(username, filename),
565:            '''stat -c '%U' /home/{0}/{1}'''.format(username, filename),
```
I mean ones in `docs`, these in `tests` are obviously false-positives.

Also, this (this matches space between `'` or `"`):
```
$  ack -l "([\"'])\s+\1(?!(\.join|\n))"
scripts/gen_api_docs.sh
robottelo/config/base.py
robottelo/decorators/func_locker.py
robottelo/cli/hammer.py
robottelo/cli/factory.py
robottelo/constants.py
robottelo/vm.py
robottelo/system_facts.py
robottelo/datafactory.py
robottelo/api/utils.py
robottelo/api/entity_fixtures.py
robottelo/helpers.py
tests/robottelo/test_datafactory.py
tests/robottelo/test_assertions.py
tests/robottelo/test_cli.py
tests/robottelo/test_hammer.py
tests/foreman/ui/test_dashboard.py
tests/foreman/ui/test_package.py
tests/foreman/ui/test_settings.py
tests/foreman/ui/test_oscappolicy.py
tests/foreman/ui/test_discoveredhost.py
tests/foreman/ui/test_activationkey.py
tests/foreman/ui/test_subscription.py
tests/foreman/ui/test_hostcollection.py
tests/foreman/ui/test_ldap_authentication.py
tests/foreman/ui/test_contentcredential.py
tests/foreman/cli/test_lifecycleenvironment.py
tests/foreman/cli/test_repository.py
tests/foreman/cli/test_syncplan.py
tests/foreman/cli/test_activationkey.py
tests/foreman/cli/test_myaccount.py
tests/foreman/cli/test_user.py
tests/foreman/cli/test_oscap_tailoringfiles.py
tests/foreman/cli/test_classparameters.py
tests/foreman/cli/test_discoveryrule.py
tests/foreman/cli/test_contentview.py
tests/foreman/cli/test_host.py
tests/foreman/cli/test_hammer.py
tests/foreman/sys/test_katello_certs_check.py
tests/foreman/ui_deprecated/test_computeresource_rhev.py
tests/foreman/longrun/test_provisioning_computeresource.py
tests/foreman/longrun/test_oscap.py
tests/foreman/api/test_audit.py
tests/foreman/api/test_repository.py
tests/foreman/api/test_syncplan.py
tests/foreman/api/test_template.py
tests/foreman/api/test_partitiontable.py
tests/foreman/api/test_docker.py
tests/foreman/api/test_classparameters.py
tests/upgrades/test_client.py
tests/upgrades/test_errata.py
tests/upgrades/test_performance_tuning.py
```

Some of these are false-positives:
```
tests/foreman/api/test_partitiontable.py
189:        for layout in ('', ' ', None):
283:        for new_layout in ('', ' ', None):
```

But many are valid:
```
tests/foreman/api/test_syncplan.py
778:            'Waiting {0} seconds to check products' ' were not synced'.format(delay / 4)
787:            'Waiting {0} seconds to check products' ' were synced'.format(delay * 3 / 4)
```

We have merge conflict already, and we are only going to have more of them. My recommendation would be to rebase in evening US time and merge right away, and raise separate PR for remaining `u` and non-combined strings.

**Comment by JacobCallahan** [comment]
@mirekdlugosz so I was debating with myself whether or not I should include docs in this change as well since I'd like to do an overall review of docs later on. If you think it is worth it to include them in this, then I can certainly update.

I can also handle the extra string spaces as well.

Yeah, merge conflicts are definitely going to be a thing, so we'll have to merge pretty quickly after we're happy with the state of this PR!

**Comment by mirekdlugosz** [comment]
> @mirekdlugosz so I was debating with myself whether or not I should include docs in this change as well since I'd like to do an overall review of docs later on. If you think it is worth it to include them in this, then I can certainly update.

Let's be reasonable. These are only two files, way too small to sway review one way or another and not significant enough to spend too much time arguing about it. Either way is fine.

Personally I would include them here, because these are only two files.

> Yeah, merge conflicts are definitely going to be a thing, so we'll have to merge pretty quickly after we're happy with the state of this PR!

I would push to merge it today, because Friday afternoon/evening US time is the best time to merge highly disruptive changes like that. Our next good window will be in a week.

**Comment by pondrejk** [comment]
pending rebase

**Comment by JacobCallahan** [comment]
@mirekdlugosz @mshriver Ok, everything has been updated based on past feedback and we're at no merge conflicts. If I don't have any objections by the end of the day, then we should merge.

**Comment by mshriver** [comment]
@JacobCallahan no objection from me!

**Pull request merged**